framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -263,47 +269,42 @@ public void updateForMethodCall(
                     FieldAccess fieldAccess = e.getKey();
                     V otherVal = e.getValue();
 
-                    // case 3: the field has a monotonic annotation
-                    if (!((GenericAnnotatedTypeFactory<?, ?, ?, ?>) atypeFactory)
-                            .getSupportedMonotonicTypeQualifiers()
-                            .isEmpty()) {
-                        List<Pair<AnnotationMirror, AnnotationMirror>> fieldAnnotations =
-                                atypeFactory.getAnnotationWithMetaAnnotation(
-                                        fieldAccess.getField(), MonotonicQualifier.class);
-                        V newOtherVal = null;
-                        for (Pair<AnnotationMirror, AnnotationMirror> fieldAnnotation :
-                                fieldAnnotations) {
-                            AnnotationMirror monotonicAnnotation = fieldAnnotation.second;
-                            @SuppressWarnings("deprecation") // permitted for use in the framework
-                            Name annotation =
-                                    AnnotationUtils.getElementValueClassName(
-                                            monotonicAnnotation, "value", false);
-                            AnnotationMirror target =
-                                    AnnotationBuilder.fromName(
-                                            atypeFactory.getElementUtils(), annotation);
-                            // Make sure the 'target' annotation is present.
-                            if (AnnotationUtils.containsSame(otherVal.getAnnotations(), target)) {
-                                newOtherVal =
-                                        analysis.createSingleAnnotationValue(
-                                                        target, otherVal.getUnderlyingType())
-                                                .mostSpecific(newOtherVal, null);
-                            }
-                        }
+                    // Case 1: The field is unassignable
+                    if (fieldAccess.isUnassignableByOtherCode()) {
+                        // Keep information.
+                        newFieldValues.put(fieldAccess, otherVal);
+                        continue;
+                    }
+
+                    // Case 2: The field has a monotonic annotation.
```

Comment:
I think the "case" numbers refer to the ordering in the method javadoc. This looks inconsistent now.